pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -554,16 +554,57 @@ func (qjm *XController) ScheduleNext() {
 	// amount of resources asked by the job
 	qj, err := qjm.qjqueue.Pop()
 	if err != nil {
-		glog.V(4).Infof("[ScheduleNext] Cannot pop QueueJob from qjqueue!")
+		glog.V(4).Infof("[ScheduleNext] Cannot pop QueueJob from qjqueue! err=%#v", err)
+		return // Try to pop qjqueue again
 	} else {
-		glog.V(4).Infof("[ScheduleNext] activeQ.Pop %s *Delay=%s RemainingLength=%d &qj=%p Version=%s Status=%+v", qj.Name, metav1.Now().Sub(qj.Status.ControllerFirstTimestamp.Time), qjm.qjqueue.Length(), qj, qj.ResourceVersion, qj.Status)
+		glog.V(4).Infof("[ScheduleNext] activeQ.Pop %s *Delay=%.6f seconds RemainingLength=%d &qj=%p Version=%s Status=%+v", qj.Name, time.Now().Sub(qj.Status.ControllerFirstTimestamp.Time).Seconds(), qjm.qjqueue.Length(), qj, qj.ResourceVersion, qj.Status)
 	}
+
+	// Re-compute SystemPriority for DynamicPriority policy
+	if qjm.serverOption.DynamicPriority {
+		//  Create newHeap to temporarily store qjqueue jobs for updating SystemPriority
+		tempQ := newHeap(cache.MetaNamespaceKeyFunc, HigherSystemPriorityQJ)
+		qj.Status.SystemPriority = qj.Spec.Priority + qj.Spec.PrioritySlope * (time.Now().Sub(qj.Status.ControllerFirstTimestamp.Time)).Seconds()
+		tempQ.Add(qj)
+		for qjm.qjqueue.Length() > 0 {
+			qjtemp, _ := qjm.qjqueue.Pop()
+			qjtemp.Status.SystemPriority = qjtemp.Spec.Priority + qjtemp.Spec.PrioritySlope * (time.Now().Sub(qjtemp.Status.ControllerFirstTimestamp.Time)).Seconds()
+//			qjtemp.Status.FilterIgnore = true   // update SystemPriority only
+//			qjm.updateEtcd(qjtemp, "[ScheduleNext]updateSystemPriority")
```

Comment:
Could you remove dead code?  Thanks!

Reply:
two lines removed.

Reply:
Per our brief discussion, I understand this code was to update etcd with the latest dynamic priority value which has some impact on performance thus the reason for commenting it out until a more desirable solution can be obtained.